### PR TITLE
Added annotations to select SSL Profile on the HTTPS listener

### PR DESCRIPTION
**Pull request by secmff**
The SSL profile must be predefined on the AppGw but can be selected by
name using the annotation: appgw-ssl-profile. Added unittests and
documentation.

## Checklist
- [x] The title of the PR is clear and informative
- [x] If applicable, the changes made in the PR have proper test coverage
- [ ] Issues addressed by the PR are mentioned in the description followed by `Fixes`.

### CODE REVIEW

**Comment by williamalvvc** [comment]
Any updates on this?

**Review by williamalvvc — APPROVED** [review]

**Comment by secmff** [comment]
> Any updates on this?

Is there anything I can do to allow progress?  I did do a rebase.

File: pkg/tests/fixtures/ssl_profile.go [hunk 1]
```diff
@@ -0,0 +1,23 @@
...
+	n "github.com/Azure/azure-sdk-for-go/services/network/mgmt/2021-03-01/network"
+	"github.com/Azure/go-autorest/autorest/to"
+)
+
+const (
+	// CertificateName1 is a string constant.
+	SslProfileName1 = "hardend-tls"
+)
+
+// GetCertificate1 generates a certificate.
```
Comment (akshaysngupta) [inline pkg/tests/fixtures/ssl_profile.go:18]: nit: GetSslProfile1

**Comment by akshaysngupta** [comment]
@secmff thanks for the PR. It looks good to me. Once you fix the comment, I will merge.

File: pkg/tests/fixtures/ssl_profile.go (hunk 1 above)
Comment (secmff) [reply pkg/tests/fixtures/ssl_profile.go]: Thanks you very much. Fixed and did a rebase.

**Comment by KittyKatt** [comment]
We're currently waiting on this PR for an implementation our client needs in their environment to specify SSL profiles on-demand with AGIC. Is there any movement on reviewing this and getting it merged?

**Comment by trentcain** [comment]
Hi @secmff @akshaysngupta. Will this PR fix #954? If so, will it also resolve the issue where AGIC is currently removing SSL Profiles from the Application Gateway?

**Comment by secmff** [comment]
I think it will fix it.  The SSL profile gets deleted because a whole new config will be loaded onto the AG. If you use the annotations from this PR, that config will include the SSL profile.  Please give it a try and let us know your results.

**Comment by secmff** [comment]
Thanks @trentcain 
I did a rebase, should be ready for merge.

**Comment by joebeeplanet** [comment]
Hi any update on this PR?

**Comment by jkroepke** [comment]
@akshaysngupta Is there a date for release which include this PR?

**Comment by acc0613** [comment]
@akshaysngupta will this be released soon? It is currently affecting our mutual authentication implementation as the SSL profile is getting removed on the listener

**Comment by klerkda** [comment]
When will this PR become part of an offical release? We are desparately waiting for it.

**Comment by klerkda** [comment]
@akshaysngupta Do you know when this will be part of an official release? My company only wants to deploy official releases and we really need this fix.

**Pull request merged**